modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/exec/DdlWithMockedManagersTest.java

```diff
@@ -0,0 +1,461 @@
+    /** Table partitions. */
+    private static final int PARTITIONS = 32;
+
+    /** Node name. */
+    private static final String NODE_NAME = "node1";
+
+    /** Schema manager. */
+    @Mock
+    private BaselineManager bm;
+
+    /** Topology service. */
+    @Mock
+    private TopologyService ts;
+
+    /** Cluster service. */
+    @Mock(lenient = true)
+    private ClusterService cs;
+
+    /** Raft manager. */
+    @Mock
+    private Loza rm;
+
+    /** TX manager. */
+    @Mock(lenient = true)
+    private TxManager tm;
+
+    /** Tables configuration. */
+    @InjectConfiguration(
+            internalExtensions = ExtendedTableConfigurationSchema.class,
+            polymorphicExtensions = {
+                    HashIndexConfigurationSchema.class, SortedIndexConfigurationSchema.class, PartialIndexConfigurationSchema.class
+            }
+    )
+
+    private TablesConfiguration tblsCfg;
+
+    /** Data storage configuration. */
+    @InjectConfiguration
+    private DataStorageConfiguration dataStorageCfg;
+
+    TableManager tblManager;
+
+    SqlQueryProcessor queryProc;
+
+    /** Test node. */
+    private final ClusterNode node = new ClusterNode(
+            UUID.randomUUID().toString(),
+            NODE_NAME,
+            new NetworkAddress("127.0.0.1", 2245)
+    );
+
+    /** Returns current method name. */
+    private static String getCurrentMethodName() {
+        return StackWalker.getInstance()
+                .walk(s -> s.skip(1).findFirst())
+                .get()
+                .getMethodName();
+    }
+
+    /** Stop configuration manager. */
+    @AfterEach
+    void after() {
+        try {
+            Objects.requireNonNull(queryProc).stop();
+        } catch (Exception e) {
+            fail(e);
+        }
+
+        Objects.requireNonNull(tblManager).stop();
+    }
+
+    /** Inner initialisation. */
+    @BeforeEach
+    void before() throws NodeStoppingException {
+        tblManager = mockManagers();
+
+        queryProc = new SqlQueryProcessor(cs, tblManager);
+
+        queryProc.start();
+    }
+
+    /**
+     * Tests create a table through public API.
+     */
+    @Test
+    public void testCreateTable() throws Exception {
+        SqlQueryProcessor finalQueryProc = queryProc;
+
+        String curMethodName = getCurrentMethodName();
+
+        String newTblSql = String.format("CREATE TABLE %s (c1 int PRIMARY KEY, c2 varbinary(255)) "
+                + "with partitions=1,replicas=1", curMethodName);
+
+        queryProc.query("PUBLIC", newTblSql);
+
+        assertTrue(tblManager.tables().stream().anyMatch(t -> t.name()
+                .equalsIgnoreCase("PUBLIC." + curMethodName)));
+
+        String finalNewTblSql1 = newTblSql;
+
+        assertThrows(TableAlreadyExistsException.class, () -> finalQueryProc.query("PUBLIC", finalNewTblSql1));
+
+        assertThrows(IgniteInternalException.class, () -> finalQueryProc.query("PUBLIC",
```

Comment:
For now it is ok.
Let's create a ticket to throw a query parsing exception here.

Reply:
ticket created.